### PR TITLE
Add IkedaCarpenterPV peak shape for PDCalibration

### PR DESCRIPTION
**Pull request by jenest**
**Description of work.**

Addresses GitLab issue [PD#275](https://code.ornl.gov/sns-hfir-scse/diffraction/powder/powder-diffraction/-/issues/275).

Version into [ornl-next](https://github.com/mantidproject/mantid/pull/31829)

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: instrument/NOMAD_Parameters.xml [hunk 1]
```diff
@@ -1,14 +1,125 @@
...
 
-<parameter name="TS_mapping_file" type="string">
-    <value val="NOM_TS_2010_12_01.dat"/>
-</parameter>
+        <parameter name="TS_mapping_file" type="string">
+            <value val="NOM_TS_2010_12_01.dat"/>
+        </parameter>
 
+        <parameter name="IkedaCarpenterPV:Kappa" type="fitting">
+            <formula eq="0.1"/>
```
Comment (rosswhitfield) [inline instrument/NOMAD_Parameters.xml:11]: I think this need to `result-unit` to work, so can you make this `<formula eq="0.1" result-unit="TOF"/>`.

File: Framework/Algorithms/test/PDCalibrationTest.h [hunk 2]
```diff
@@ -475,6 +477,104 @@ class PDCalibrationTest : public CxxTest::TestSuite {
...
+
+    MatrixWorkspace_const_sptr ws = AnalysisDataService::Instance().retrieveWS<MatrixWorkspace>("ws");
+    TS_ASSERT(ws);
+
+    MoveInstrumentComponent movealg;
+    TS_ASSERT_THROWS_NOTHING(movealg.initialize());
+    TS_ASSERT(movealg.isInitialized());
+    TS_ASSERT_THROWS_NOTHING(movealg.setProperty("Workspace", "ws"));
+    TS_ASSERT_THROWS_NOTHING(movealg.setPropertyValue("ComponentName", "bank1"));
+    TS_ASSERT_THROWS_NOTHING(movealg.setProperty("X", 1.0));
```
Comment (rosswhitfield) [inline Framework/Algorithms/test/PDCalibrationTest.h:553]: So this is the already perfectly calibrated positions that directly corresponds to `difc = 2208.287616521762`, can we change this so X=1.01 and Z=1.01, that will give a starting difc of 2211.02 and we check that is still get the correct original difc.

File: Framework/Algorithms/test/PDCalibrationTest.h [hunk 3]
```diff
@@ -475,6 +477,104 @@ class PDCalibrationTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(calTable->cell<double>(index + 2, 1), calTable->cell<double>(index, 1), 1E-5); // det 102
     TS_ASSERT_DELTA(calTable->cell<double>(index + 3, 1), calTable->cell<double>(index, 1), 1E-5); // det 103
   }
+
+  void test_exec_ikeda_carpenter() {
+    // test the algorithm using the IkedaCarpenterPV peak function
+    const double ref_difc = 2208.287616521762;
+
+    const std::vector<double> dValues{0.3117, 0.3257, 0.3499, 0.4205, 0.4645, 0.4768, 0.4996, 0.5150, 0.5441, 0.5642,
```
Comment (rosswhitfield) [inline Framework/Algorithms/test/PDCalibrationTest.h:485]: We can probably just do the last four peaks, `dvalues = (.8920,1.0758,1.2615,2.0599)`

File: Framework/Algorithms/test/PDCalibrationTest.h [hunk 4]
```diff
@@ -475,6 +477,104 @@ class PDCalibrationTest : public CxxTest::TestSuite {
...
+                               "    <parameter name=\"IkedaCarpenterPV:SigmaSquared\" type=\"fitting\">\n"
+                               "        <formula eq=\"1.0\" unit=\"dSpacing\" result-unit=\"TOF^2\"/>\n"
+                               "    </parameter>\n"
+                               "    <parameter name=\"IkedaCarpenterPV:Gamma\" type=\"fitting\">\n"
+                               "        <formula eq=\"1.0\" unit=\"dSpacing\" result-unit=\"TOF\"/>\n"
+                               "    </parameter>\n"
+                               "</component-link>\n"
+                               "</parameter-file>";
+
+    LoadParameterFile paramalg;
```
Comment (rosswhitfield) [inline Framework/Algorithms/test/PDCalibrationTest.h:537]: Setting this shouldn't be required as these should all be the default values anyway.

**Review by rosswhitfield — CHANGES_REQUESTED** [review]
Small changes to make the test a better one. In python this seems to work

```python
dvalues = (.8920,1.0758,1.2615,2.0599)
expected_difc = 2208.287616521762
myFunc = ';'.join(f"name=IkedaCarpenterPV, X0={expected_difc*dvalue}, I=50" for dvalue in dvalues)
ws = CreateSampleWorkspace("Event","User Defined",myFunc, XMin=1, XMax=16666.7, BinWidth=1, NumEvents=100000, BankPixelWidth=1, NumBanks=1)
MoveInstrumentComponent(ws, ComponentName='bank1', X=1.01, Y=0, Z=1.01, RelativePosition=False)
PDCalibration(ws, TofBinning='1,1,16666', PeakFunction='IkedaCarpenterPV', PeakPositions=dvalues, OutputCalibrationTable='cal', DiagnosticWorkspaces='diag')
print(mtd['cal'].column('difc')[0])
```


Add to the v6.2.0 diffraction release notes that PDCalibration has the new peak type.

File: Framework/Algorithms/test/PDCalibrationTest.h (hunk 2 above)
Comment (ckendrick) [reply Framework/Algorithms/test/PDCalibrationTest.h]: It is now giving a DIFC of around 2208.31 with the new instrument shift which seems to be good

**Review by rosswhitfield — DISMISSED** [review]

**Review by rosswhitfield — APPROVED** [review]

**Pull request merged**